internal/pkg/agent/storage/encrypted_disk_store.go

```diff
@@ -46,11 +47,21 @@ func NewEncryptedDiskStore(ctx context.Context, target string, opts ...OptionFun
 	if encryptionDisabled {
 		return NewDiskStore(target)
 	}
+
+	unprivileged := false
+
+	hasRoot, err := utils.HasRoot()
+	if err != nil || !hasRoot {
+		// TODO log error
```

Comment:
TODO here?

Reply:
Propagated the error in https://github.com/elastic/elastic-agent/pull/4362/files/6b2dc5973182afc60dd44aad888f291e57c3adba..78425bf5734c815486b57c4430bb5e6c0daf642a